Add io.github.benladamm.ssh-client

benladamm opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [x] Please describe the application briefly.
Modern SSH Client is a lightweight, native GTK4 application designed for the GNOME desktop. It provides a clean interface for managing SSH connections and features a fully functional terminal emulator based on VTE.
- [x] Please attach a video showcasing the application on Linux using the Flatpak. [Video](https://github.com/user-attachments/assets/0acc4157-9674-423f-8234-452e656b2745)
- [x] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [x] I am an _(please keep whichever is applicable and remove the rest)_ author/developer/upstream contributor to the project.
      If not, I contacted upstream developers about this submission. **Link:**

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

benladamm committed:
Adding yaml file

hfiguiere commented on io.github.benladamm.ssh_client.yml:
```diff
@@ -0,0 +1,50 @@
+app-id: io.github.benladamm.ssh_client
+runtime: org.gnome.Platform
+runtime-version: '47'
```
this runtime is EOL

benladamm committed:
Update runtime-version to 49

benladamm committed:
Update io.github.benladamm.ssh_client.yml

benladamm committed:
update id

benladamm committed:
Update io.github.benladamm.ssh-client.yml

benladamm commented:
sad

benladamm committed:
Update io.github.benladamm.ssh-client.yml

benladamm committed:
Update io.github.benladamm.ssh-client.yml

benladamm committed:
Update io.github.benladamm.ssh-client.yml

benladamm commented:
bot, build

benladamm committed:
Update io.github.benladamm.ssh-client.yml

hfiguiere reviewed (changes requested):
missing cleanup rules to remove the development only files.

benladamm committed:
Add cleanup steps to all modules in build config

Introduced cleanup directives for libssh, vte, and modern_ssh modules in io.github.benladamm.ssh-client.yml to remove unnecessary files and directories after build. Also updated the modern_ssh source commit to 8f382f4.

benladamm committed:
Move cleanup directives to top-level and update app commit

Consolidated repeated cleanup directives from individual modules to a top-level 'cleanup' section in the Flatpak manifest for better maintainability. Also updated the 'modern_ssh' module to use a newer commit.

benladamm commented:
bot, build

bbhtt commented on io.github.benladamm.ssh-client.yml:
```diff
@@ -0,0 +1,60 @@
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --talk-name=org.freedesktop.secrets
```
why does it not use the secrets portal?

benladamm committed:
Update io.github.benladamm.ssh-client.yml

Co-authored-by: bbhtt <bbhtt.zn0i8@slmail.me>
